src/Components/Modals/DeleteAdditionalRepositoriesModal.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import {
+  Button,
+  Modal,
+  ModalVariant,
+  TextContent,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons';
+import propTypes from 'prop-types';
+import useNotifications from '../../hooks/useNotifications';
+import { useQueryClient } from 'react-query';
+import useDeleteAdditionalRepositories from '../../hooks/useDeleteAdditionalRepositories';
+
+const DeleteAdditionalRepositoriesModal = (props) => {
+  const {
+    isOpen,
+    handleModalToggle,
+    name,
+    repositoryNameToDelete,
+    repositoryLabelToDelete,
+  } = props;
+  const { addSuccessNotification, addErrorNotification } = useNotifications();
+  const { mutate, isLoading } = useDeleteAdditionalRepositories();
+  const queryClient = useQueryClient();
+
+  const deleteAdditionalRepositories = (
+    name,
+    repositoryNameToDelete,
+    repositoryLabelToDelete
+  ) => {
+    const payload = [
+      {
+        repositoryLabel: repositoryLabelToDelete,
+        repositoryName: repositoryNameToDelete,
+      },
+    ];
+
+    mutate(
+      { name, payload },
+      {
+        onSuccess: (data, queryName) => {
+          const updatedData = data?.filter(
+            (entry) => entry.repositoryName !== repositoryNameToDelete
+          );
+          queryClient.setQueryData(queryName, updatedData);
+          addSuccessNotification(
+            `Additional repository ${repositoryNameToDelete} deleted`
+          );
+          queryClient.invalidateQueries(queryName);
```

Comment:
We don't need to both `setQueryData` and `invalidateQueries` - just one or the other